builtin/credential/okta/path_config.go

```diff
@@ -153,22 +158,18 @@ func (b *backend) pathConfigWrite(
 		return logical.ErrorResponse("api_token is missing"), nil
 	}
 
-	baseURL, ok := d.GetOk("base_url")
-	if ok {
-		baseURLString := baseURL.(string)
-		if len(baseURLString) != 0 {
-			_, err = url.Parse(baseURLString)
-			if err != nil {
-				return logical.ErrorResponse(fmt.Sprintf("Error parsing given base_url: %s", err)), nil
-			}
-			cfg.BaseURL = baseURLString
-		}
-	} else if req.Operation == logical.CreateOperation {
-		cfg.BaseURL = d.Get("base_url").(string)
+	baseURL := d.Get("base_url").(string)
+	_, err = url.Parse(fmt.Sprintf("https://%s,%s", cfg.Org, baseURL))
+	if err != nil {
+		return logical.ErrorResponse(fmt.Sprintf("Error parsing given base_url: %s", err)), nil
 	}
+	cfg.BaseURL = baseURL
 
-	productionRaw := d.Get("production").(bool)
-	cfg.Production = &productionRaw
+	productionRaw, ok := d.GetOk("production")
+	if ok {
+		production := productionRaw.(bool)
+		cfg.Production = &production
```

Comment:
Because this loads cfg first if it's there, if they have production in the past, but now leave that value out and switch to a custom base url, there's a possibility of their base url being overwritten by the production value later on. I think if a base_url is provided we should nil out production.

Reply:
Sounds good.  I played around with this dance a bit and ended up hear.   I like your suggestion better.